src/NetMQ/NetMQSocket.cs

```diff
@@ -343,18 +343,10 @@ internal void InvokeEvents(object sender, PollEvents events)
             m_socketEventArgs.Init(events);
 
             if (events.HasIn())
-            {
-                var temp = m_receiveReady;
-                if (temp != null)
-                    temp(sender, m_socketEventArgs);
-            }
+                m_receiveReady?.Invoke(sender, m_socketEventArgs);
```

Comment:
Null propagation does a copy first. Identical to the code you have there. 


Sent from my Samsung Galaxy smartphone.-------- Original message --------From: Doron Somech <notifications@github.com> Date: 31/03/2016  18:25  (GMT+08:00) To: zeromq/netmq <netmq@noreply.github.com> Cc: Calin <pcalin@yahoo.com> Subject: Re: [zeromq/netmq] Start using C# 6 language features (#522) 
In src/NetMQ/NetMQSocket.cs:
> @@ -343,18 +343,10 @@ internal void InvokeEvents(object sender, PollEvents events)
>              m_socketEventArgs.Init(events);
>  
>              if (events.HasIn())
> -            {
> -                var temp = m_receiveReady;
> -                if (temp != null)
> -                    temp(sender, m_socketEventArgs);
> -            }
> +                m_receiveReady?.Invoke(sender, m_socketEventArgs);

I'm not sure how null propagation is working, but this might be not safe if null propagation is not copying the variable before the null check. the temp variable is used to avoid the race condition. 

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub